### ISSUE TITLE
training speed about fairseq

### ISSUE THREAD

**Issue by hchen1024** [pytorch/fairseq#3098]
## ❓ Questions and Help
On the machine A, wps=120000, on another machine B, wps=60000, what's wrong about this?

# What's your environment?

 - fairseq Version (e.g., 1.0 or master): 0.10.1（A and B）
 - PyTorch Version (e.g., 1.0)：1.7.1 （A）1.7.0（B）
 - OS (e.g., Linux): Linux Ubuntu16.04
 - How you installed fairseq (`pip`, source): pip
 - Build command you used (if compiling from source): 
 - Python version: 3.7.5
 - CUDA/cuDNN version: 11.0(A) 10.2(B)
 - GPU models and configuration: transformer, 6gpus
 - Any other relevant information:

**Comment by myleott** [comment]
It seems like the only difference is PyTorch 1.7.0 and PyTorch 1.7.1? Perhaps there's some improvement in 1.7.1 that makes training faster?

Without more details about your setup/commands run, etc., it's impossible to say for sure :)

**Comment by ansong1024** [comment]
> It seems like the only difference is PyTorch 1.7.0 and PyTorch 1.7.1? Perhaps there's some improvement in 1.7.1 that makes training faster?
> 
> Without more details about your setup/commands run, etc., it's impossible to say for sure :)
command:
python -m torch.distributed.launch --nproc_per_node=6 --nnodes=1 --node_rank=0 --master_addr=127.0.0.1 --master_port=16123 --use_env train_v0.10.py --task speech_recognition --arch vggtransf_conf_1 --criterion cross_entropy_with_acc --optimizer adam --lr-scheduler noam --log-interval 100 --log-format simple --seed 23424 --user-dir asr_transf_conf --vocab-threshold 3 --lmdb-dir chs_dataset --mean-npy mean.npy --std-npy std.npy --feat-dim 80 --max-frames 16700 --min-frame-num 200 --max-frame-num 1600 --required-sample-size-multiple 8 --chunk-size 1000 --specaug-T 100 --specaug-F 20 --char-vocab train_units.dict --num-workers 0 --train-subset train --valid-subset dev --batch-size 83 --required-batch-size-multiple 8 --distributed-world-size 6 --max-epoch 20 --max-update 350000 --clip-norm 1.0 --update-freq 1 --lr 0.0002 --min-lr 0.000001 --tensorboard-logdir tensorboard --save-interval-updates 5000 --adam-betas "(0.9,0.98)" --weight-decay 0.00005 --warmup-updates 20000 --label-smoothing 0.1 --lr-alpha 0.03 --lr-shrink 0.3 --milestones 320000,340000 --fp16-init-scale 128 --fp16-scale-window 2000 --threshold-loss-scale 1.0 --no-progress-bar --specaug --fp16 --fix-batches-to-gpus --find-unused-parameters --save-dir checkpoints --distributed-no-spawn --ddp-backend c10d --distributed-backend nccl

log: 
machine A       2021-01-04 13:53:41 | INFO | train_inner | epoch 001:    300 / 12377139 loss=0.88, ce_loss=1.139, nll_loss=0.96, acc=0.83, wps=110333, ups=1.2, wpb=92139.7, bsz=283.6, num_updates=100300, lr=9.47263e-05, gnorm=0.911, clip=8, loss_scale=2048, train_wall=70, wall=0
machine B.      2021-01-04 21:56:30 | INFO | train_inner | epoch 001:    300 / 12377139 loss=0.881, ce_loss=1.134, nll_loss=0.952, acc=0.832, wps=61846.5, ups=0.68, wpb=90680.1, bsz=266.3, num_updates=100300, lr=9.47263e-05, gnorm=0.933, clip=8, loss_scale=2048, train_wall=144, wall=0

**Comment by ansong1024** [comment]
@myleott

**Comment by ansong1024** [comment]
@myleott
